Userland/Shell/Builtin.cpp

```diff
@@ -729,16 +725,10 @@ int Shell::builtin_pushd(int argc, const char** argv)
         }
     }
 
-    LexicalPath lexical_path(path_builder.to_string());
-    if (!lexical_path.is_valid()) {
-        warnln("LexicalPath failed to canonicalize '{}'", path_builder.string_view());
-        return 1;
-    }
-
-    const char* real_path = lexical_path.string().characters();
+    auto real_path = LexicalPath::canonicalized_path(path_builder.to_string());
```

Comment:
That's not a very real real_path, but that doesn't have anything to do with this PR :P